### PR TITLE
feat(#1602): mark with versions

### PR DESCRIPTION
**Pull request by maxonfjvipon**
Ref: #1602

What's done:
1. Added two more constructors for `OnReplaced`
2. Added and refactored tests for `OnReplaced`
3. Added ability to extend tojos via `MarkMojo` with versions
4. Added tests for `MarkMojo`

---

## PR-Codex overview
This PR focuses on improving the code readability and consistency in the `eo-maven-plugin` module. 

### Detailed summary
- Updated comments and method names for clarity and consistency.
- Improved variable naming and formatting.
- Added missing imports and dependencies.
- Refactored code to use more descriptive class and variable names.
- Added unit tests for improved code coverage.

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

**Comment by maxonfjvipon** [comment]
@volodya-lombrozo please review this one

File: eo-maven-plugin/src/main/java/org/eolang/maven/MarkMojo.java [hunk 1]
```diff
@@ -78,11 +81,22 @@ public void exec() throws IOException {
     private long scan(final Path dir, final String version) {
         final Unplace unplace = new Unplace(dir);
         final Collection<Path> sources = new Walk(dir);
-        final ForeignTojos scoped = this.scopedTojos();
+        final ForeignTojos tojos = this.scopedTojos();
         final long done = sources.stream()
             .filter(src -> src.toString().endsWith(".eo"))
-            .map(unplace::make)
-            .map(scoped::add)
+            .map(
```
Comment (volodya-lombrozo) [inline eo-maven-plugin/src/main/java/org/eolang/maven/MarkMojo.java:87]: @maxonfjvipon [optional suggestion] We have lots indents here and code looks a bit [complicated](https://en.wikipedia.org/wiki/Cyclomatic_complexity) now.  Maybe it's better to move it in a separate method with meaningful name?

File: eo-maven-plugin/src/main/java/org/eolang/maven/ProbeMojo.java [hunk 2]
```diff
@@ -52,7 +52,7 @@
 
 /**
  * Go through all `probe` metas in XMIR files, try to locate the
- * objects pointed by `probe` in Objectionary and if found register them in
+ * objects pointed by `probe` in Objectionary and if found, register them in
```
Comment (volodya-lombrozo) [inline eo-maven-plugin/src/main/java/org/eolang/maven/ProbeMojo.java:55]: @maxonfjvipon 
```
...try to locate the objects pointed to by `probe` in Objectionary, and if found, register them in the catalog
```
I added one more comma and `the` article.

File: eo-maven-plugin/src/main/java/org/eolang/maven/name/OnReplaced.java [hunk 3]
```diff
@@ -67,13 +70,33 @@ public final class OnReplaced implements ObjectName {
      * - "org.eolang.string#1.23.1"
      * - "org.eolang.math#3.3.3"
      */
-    private final String raw;
+    private final Unchecked<String> raw;
```
Comment (volodya-lombrozo) [inline eo-maven-plugin/src/main/java/org/eolang/maven/name/OnReplaced.java:73]: @maxonfjvipon I don't see the reason for that change. Can we keep that code [simple](https://en.wikipedia.org/wiki/KISS_principle)?

File: eo-maven-plugin/src/test/java/org/eolang/maven/MarkMojoTest.java [hunk 4]
```diff
@@ -65,16 +71,45 @@ void updatesVersionIfItExists(@TempDir final Path temp) throws IOException {
...
+                "Tojos should have contained versioned object %s after extending, but they didn't",
+                object
+            ),
+            tojos.contains(object),
+            Matchers.is(true)
+        );
+    }
+
+    /**
+     * Save foo/bar.eo file to target/hello/-/0.28.0/EO-SOURCES.
```
Comment (volodya-lombrozo) [inline eo-maven-plugin/src/test/java/org/eolang/maven/MarkMojoTest.java:101]: @maxonfjvipon It is dangerous to duplicate logic in comments, as the logic might change without corresponding updates to the comments. Moreover, why should we state the same thing twice? The code already expresses what is going on clearly.

**Review by volodya-lombrozo — CHANGES_REQUESTED** [review]

File: eo-maven-plugin/src/main/java/org/eolang/maven/MarkMojo.java [hunk 5]
```diff
@@ -92,4 +96,23 @@ private long scan(final Path dir, final String version) {
...
+     * Convert given file to object name.
+     * @param file File name
+     * @param unplace Unplace for given file
+     * @param version Version
+     * @return Object name from given file
+     */
+    private ObjectName objectName(final Path file, final Unplace unplace, final String version) {
+        return new OnSwap(
+            this.withVersions,
+            new OnReplaced(
```
Comment (maxonfjvipon) [inline eo-maven-plugin/src/main/java/org/eolang/maven/MarkMojo.java:109]: @volodya-lombrozo here

File: eo-maven-plugin/src/main/java/org/eolang/maven/name/OnReplaced.java (hunk 3 above)
Comment (maxonfjvipon) [reply eo-maven-plugin/src/main/java/org/eolang/maven/name/OnReplaced.java:73]: @volodya-lombrozo please check code [here](https://github.com/objectionary/eo/pull/2398/files#r1294913645)

**Comment by maxonfjvipon** [comment]
@volodya-lombrozo please have a look one more time

**Review by volodya-lombrozo — APPROVED** [review]
@maxonfjvipon Looks good to me, thanks

**Comment by maxonfjvipon** [comment]
@yegor256 please have a look

**Pull request merged**